sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -184,30 +190,62 @@ impl InputSelection {
             return Err(Error::MissingInputWithEd25519Address);
         };
 
-        let mut remainder_builder = BasicOutputBuilder::new_with_amount(amount_diff).with_mana(mana_diff);
+        let mut remainder_outputs = Vec::new();
 
-        remainder_builder =
-            remainder_builder.add_unlock_condition(AddressUnlockCondition::new(remainder_address.clone()));
-
-        // TODO https://github.com/iotaledger/iota-sdk/issues/1631
-        // TODO Only putting one in remainder atm so we can at least create foundries
         if let Some(native_tokens) = native_tokens_diff {
-            log::debug!("Adding {native_tokens:?} to remainder output for {remainder_address:?}");
-            remainder_builder = remainder_builder.with_native_token(*native_tokens.first().unwrap());
-        }
-
-        let remainder = remainder_builder.finish_output()?;
+            let last_index = native_tokens.len() - 1;
+            let mut last_output_amount = amount_diff;
+            // Create a remainder output with minimum amount for each native token and put remaining amount + mana in
+            // the last one.
+            for (n, native_token) in native_tokens.into_iter().enumerate() {
+                let remainder_builder = if n < last_index {
+                    BasicOutputBuilder::new_with_minimum_amount(self.protocol_parameters.storage_score_parameters())
+                } else {
+                    // All remainder mana in the last remainder output which also gets all remaining amount.
+                    BasicOutputBuilder::new_with_amount(last_output_amount).with_mana(mana_diff)
+                };
 
-        log::debug!("Created remainder output of amount {amount_diff} and mana {mana_diff} for {remainder_address:?}");
+                let remainder = remainder_builder
+                    .add_unlock_condition(AddressUnlockCondition::new(remainder_address.clone()))
+                    .with_native_token(native_token)
+                    .finish_output()?;
 
-        remainder.verify_storage_deposit(self.protocol_parameters.storage_score_parameters())?;
+                if n < last_index {
+                    last_output_amount = last_output_amount.saturating_sub(remainder.amount());
```

Comment:
why `saturating_sub`? can this loop continue on in a valid fashion if `last_output_amount` becomes `0`?

Reply:
If there is no error in the calculation, then no
But if it would become 0 it would then just fail when creating the output